### PR TITLE
Anfield: Support for Active Directory and Qstar target.

### PR DESCRIPTION
**Pull request by naveena-maplelabs**
1) Export AD sources from one cluster to another.
2) Export/Import active directory entries and AD objects
along with roles

### CODE REVIEW

File: samples/project-anfield/export_cluster_config.py [hunk 1]
```diff
@@ -84,7 +85,10 @@
     "sources": library.get_protection_sources(cohesity_client),
     "remote_clusters": library.get_remote_clusters(cohesity_client),
     "sql_entity_mapping": library.get_sql_entity_mapping(cohesity_client,
-                                                         env_enum.KSQL)
+                                                         env_enum.KSQL),
+    "ad": library.get_ad_entries(cohesity_client),
```
Comment (pyashish) [inline samples/project-anfield/export_cluster_config.py:89]: Can this be config driven ? Selectively import AD only if specified in the config ?

File: samples/project-anfield/export_cluster_config.py [hunk 2]
```diff
@@ -84,7 +85,10 @@
...
+    "ad_objects": library.get_ad_objects(cohesity_client),
+    "roles": cohesity_client.roles.get_roles()
```
Comment (pyashish) [inline samples/project-anfield/export_cluster_config.py:91]: Same with roles.

File: samples/project-anfield/export_cluster_config.py [hunk 3]
```diff
@@ -84,7 +85,10 @@
     "sources": library.get_protection_sources(cohesity_client),
     "remote_clusters": library.get_remote_clusters(cohesity_client),
     "sql_entity_mapping": library.get_sql_entity_mapping(cohesity_client,
-                                                         env_enum.KSQL)
+                                                         env_enum.KSQL),
+    "ad": library.get_ad_entries(cohesity_client),
+    "ad_objects": library.get_ad_objects(cohesity_client),
```
Comment (pyashish) [inline samples/project-anfield/export_cluster_config.py:90]: This would be part of sources right ?

Comment (pyashish) [reply samples/project-anfield/export_cluster_config.py:90]: And let this be by default unlike roles/ AD as access management.

**Review by pyashish — CHANGES_REQUESTED** [review]

File: samples/project-anfield/export_cluster_config.py (hunk 2 above)
Comment (naveena-maplelabs) [reply samples/project-anfield/export_cluster_config.py]: okay

File: samples/project-anfield/export_cluster_config.py (hunk 1 above)
Comment (naveena-maplelabs) [reply samples/project-anfield/export_cluster_config.py]: Updated to fetch from config file

File: samples/project-anfield/config.ini [hunk 4]
```diff
@@ -51,3 +54,6 @@ db_password=<database_password>
 username=<ad_user>
 password=<ad_password>
 machine_account=<list of comma separated values> #Optional
+
+[Qstar External Target]
```
Comment (pyashish) [inline samples/project-anfield/config.ini:58]: We capture Username and Vol ?

File: samples/project-anfield/export_cluster_config.py [hunk 5]
```diff
@@ -70,8 +72,14 @@
 logger.info("Exporting resources from cluster '%s'" % (
     configparser.get('export_cluster_config', 'cluster_ip')))
 
-# Skip paused jobs and failover ready jobs by setting this flag to true in config file.
-skip_jobs = configparser.getboolean('export_cluster_config', 'skip_jobs')
+try:
```
Comment (pyashish) [inline samples/project-anfield/export_cluster_config.py:75]: Good catch, can you add this section in netbackup project as well?

File: samples/project-anfield/import_cluster_config.py [hunk 6]
```diff
@@ -141,6 +150,22 @@ def create_vaults():
         if vault.name in available_vaults_dict.keys():
             imported_res_dict["External Targets"].append(vault.name)
             continue
+        if vault.config.qstar: # Qstar target.
+            try:
+                body = Vault()
```
Comment (pyashish) [inline samples/project-anfield/import_cluster_config.py:155]: Did you test the qstar being part of a policy and export/import the policy after the qstar has been added ?

**Review by pyashish — APPROVED** [review]

File: samples/project-anfield/config.ini (hunk 4 above)
Comment (naveena-maplelabs) [reply samples/project-anfield/config.ini:58]: Yeah

**Pull request merged**